OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/ReflectionRelatedFieldAccessesAnalysis.scala

```diff
@@ -362,15 +345,18 @@ sealed trait FieldInstanceBasedReflectiveFieldAccessAnalysis extends ReflectionA
                         () => failure(accessPC, actualReceiver, actualParameter, matchers)
                     )
 
-                if (!matchers.contains(NoFieldsMatcher))
-                    matchers += constructClassBasedFieldMatcherForClassVar(receiver.asVar, matchers, !isGetField)
-
-            case call @ VirtualFunctionCall(_, ObjectType.Class, _, "getDeclaredFields" | "getFields", _, receiver, _) =>
-                val isGetFields = call.name == "getFields"
-                if (isGetFields)
-                    matchers += PublicFieldMatcher
-                if (!matchers.contains(NoFieldsMatcher))
-                    matchers += constructClassBasedFieldMatcherForClassVar(receiver.asVar, matchers, !isGetFields)
+                if (!matchers.contains(NoFieldsMatcher)) {
```

Comment:
Should this be extracted to a new method to reuse in handleFieldArray?

Reply:
Well in the old extracted function (which got removed again when splitting this function into `handleGetField` and `handleFieldArray`), we could use the surrounding information and did not have to pass everything into the function itself. 

Imo pulling this out into a function call is not worth, as it barely reduces the amount of parameters we have to pass.

Reply:
I'm less concerned with saving space than with ensuring that everything is handled consistently and that future changes don't break the analysis because one changed one location but not the other. That being said, I'm not sure whether extracting this to a method would actually help, thus the question. If you don't see a benefit in this, I'm fine with it.